src/module.rs

```diff
@@ -73,6 +74,9 @@ pub struct Module<'a> {
 
     /// The collection of segments that compose this module.
     pub segments: Vec<Segment>,
+
+    /// the time it took to compute this module
+    pub duration: Option<Duration>,
```

Comment:
I think this could be plain `Duration`, initiliazed as `Duration::default()` (0) when it is not set yet.

Reply:
done